libtenzir/builtins/operators/from_load_read.cpp

```diff
@@ -170,6 +170,23 @@ auto parse_default_parser(std::string definition)
     .throw_();
 }
 
+auto get_loader(parser_interface& p, const char* usage, const char* docs)
+  -> std::unique_ptr<plugin_loader> {
+  auto l_name = p.accept_shell_arg();
+  if (not l_name) {
+    diagnostic::error("expected loader name")
+      .primary(p.current_span())
+      .usage(usage)
+      .docs(docs)
+      .throw_();
+  }
+  auto [loader, name] = detail::resolve_loader(p, *l_name);
+  if (not loader) {
+    throw_loader_not_found(name);
+  }
```

Comment:
My Arrow installation does not have the component for the `gcs` plugin right now (so I can't easily check it), but I think this might create a very confusing error message for `load gcs://...`:

~~~
error: loader `gcs` could not be found
 --> <input>:1:6
  |
1 | load gcs://...
  |      ^^^ 
  |
  = hint: must be one of ..., gcs, ...
  = docs: https://docs.tenzir.com/next/connectors
~~~

Perhaps you need a dedicated error path for this.

Reply:
I modified the logic in `throw_loader_not_found` and `throw_saver_not_found`: I think the error messages are better now.

Reply:
Perhaps it's even better if you replace ``loader `{}` could not be found`` with ``loader for `gs` scheme could not be found`` or something like that.

Reply:
Modified again, also to take uri-ness properly into account.